Make forced eager fallback optional in codegen

wconstab opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #75274
* #75267
* #75264

- default to generating forced fallback for TS backend (where it is used
for tests/debugging, but false otherwise

Differential Revision: [D35411211](https://our.internmc.facebook.com/intern/diff/D35411211)

wconstab committed:
Make forced eager fallback optional in codegen

- default to generating forced fallback for TS backend (where it is used
for tests/debugging, but false otherwise

[ghstack-poisoned]

JackCaoG reviewed (approved):
Thanks!

antoniojkim reviewed (approved):
Thanks for this!

bdhirsh commented on tools/codegen/gen_lazy_tensor.py:
```diff
@@ -129,7 +129,8 @@ def run_gen_lazy_tensor(aten_path: str, source_yaml: str, output_dir: str,
                         per_operator_headers: bool = False,
-                        backend_name: str = default_args.backend_name) -> None:
+                        backend_name: str = default_args.backend_name,
+                        gen_forced_fallback_code: bool = False) -> None:
```
mostly just to make sure i'm following along - doesn't this path only get taken at runtime if an environment variable is set? I'm wondering if there's any reason to make it conditional in the codegen instead of just being conditional on the env variable.

wconstab replied:
I agree with you, but Jack was wanting cleaner generated code; I think we could convince him to accept it, but there was one particular issue:

the header `ts_eager_fallback.h` needed to be included if that code is generated, so using this flag in the codegen lets us skip including that header.  If in the future we make the fallback part of the backend-api, we could consider always generating this codepath.

bdhirsh replied:
ok, sounds reasonable. The delta is super small anyway.

wconstab commented:
@wconstab has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35411211).

wconstab committed:
Update on "Make forced eager fallback optional in codegen"


- default to generating forced fallback for TS backend (where it is used
for tests/debugging, but false otherwise

Differential Revision: [D35411211](https://our.internmc.facebook.com/intern/diff/D35411211)

[ghstack-poisoned]

wconstab commented:
@wconstab has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35411211).

wconstab committed:
Update on "Make forced eager fallback optional in codegen"


- default to generating forced fallback for TS backend (where it is used
for tests/debugging, but false otherwise

Differential Revision: [D35411211](https://our.internmc.facebook.com/intern/diff/D35411211)

[ghstack-poisoned]

wconstab commented:
@wconstab has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35411211).
